azure-cs-aks-helm/MyCluster.cs

```diff
@@ -0,0 +1,89 @@
+// Copyright 2016-2021, Pulumi Corporation.  All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+using Pulumi.AzureAD;
+using Pulumi.AzureNative.ContainerService;
+using Pulumi.AzureNative.ContainerService.Inputs;
+using Pulumi.AzureNative.Resources;
+
+class MyCluster {
+    public Pulumi.Output<string> ClusterName { get; set; }
+    public Pulumi.Output<string> Kubeconfig { get; set; }
+
+    public Pulumi.Kubernetes.Provider Provider { get; set; }
+
+    public MyCluster(MyConfig cfg) {
+        var resourceGroup = new ResourceGroup("rg");
+
+        var adApp = new Pulumi.AzureAD.Application("app");
+
+        var adSp = new ServicePrincipal("service-principal", new ServicePrincipalArgs() {
+            ApplicationId = adApp.ApplicationId
+        });
+
+        var adSpPassword = new ServicePrincipalPassword("sp-password", new ServicePrincipalPasswordArgs() {
+            ServicePrincipalId = adSp.Id,
+            Value = cfg.Password,
+            EndDate = "2099-01-01T00:00:00Z"
+        });
+
+        var k8sCluster = new ManagedCluster("cluster", new ManagedClusterArgs() {
+            ResourceGroupName = resourceGroup.Name,
+            AddonProfiles = new Dictionary<string,ManagedClusterAddonProfileArgs>{
+                ["KubeDashboard"] = new ManagedClusterAddonProfileArgs() { Enabled = true }
+	        },
+ 	        AgentPoolProfiles = new ManagedClusterAgentPoolProfileArgs[]{
```

Comment:
You can use collection initializers

```suggestion
 	        AgentPoolProfiles =
 	        {
```